src/macro-context.js

```diff
@@ -139,8 +140,10 @@ ctx :: {
 */
 export default class MacroContext {
   constructor(enf, name, context, useScope, introducedScope) {
-    // todo: perhaps replace with a symbol to keep mostly private?
-    this._enf = enf;
+    privateData.set(this, { backup: enf });
+    this.reset(); // instantiate enforester
+
+    //TODO: add more fields to privateData?
```

Comment:
I just threw all the members in there. Couldn't find a reason not to. Is that cool?